core/src/main/java/nu/yona/server/analysis/entities/DayActivity.java

```diff
@@ -215,16 +187,47 @@ private int getSpreadIndex(ZonedDateTime atTime)
 	@Override
 	public void computeAggregates()
 	{
+		activityIntervals = null; // Ensure blank slate
 		totalMinutesBeyondGoal = computeTotalMinutesBeyondGoal();
 		goalAccomplished = computeGoalAccomplished();
 
 		super.computeAggregates();
+		activityIntervals = null; // Free up memory
+	}
+
+	private List<ActivityInterval> getActivityIntervals()
+	{
+		if (activityIntervals == null)
+		{
+			List<Activity> sortedActivities = activities.stream()
+					.sorted((a1, a2) -> a1.getStartTime().compareTo(a2.getStartTime())).collect(Collectors.toList());
+			activityIntervals = determineNonoverlappingActivityIntervals(sortedActivities);
+		}
+		return activityIntervals;
+	}
+
+	private static List<ActivityInterval> determineNonoverlappingActivityIntervals(List<Activity> sortedActivities)
```

Comment:
Maybe good to add a comment that activities of different apps in the same activity category, or network activity and app activity may overlap

Reply:
Makes sense.